pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -182,6 +155,106 @@ final class Event {
           'name': name,
         };
 
+  /// Event to capture previous custom dimension values from
+  /// legacy flutter analytics.
+  ///
+  /// The parameters below are a superset of all the implementations
+  /// of the `usageValues` getter in the `FlutterCommand` class within
+  /// the flutter-tool. There should never be a time where all of the parameters
+  /// are passed to this constructor.
+  ///
+  /// The custom dimensions from each implementation are grouped by sharing
+  /// the same prefix. The [workflow] indicates which implementation is being
+  /// sent, for example, "create".
+  Event.commandUsageValues({
+    required String workflow,
+    required bool commandHasTerminal,
+
+    // Assemble && build bundle implementation parameters
+    String? buildBundleTargetPlatform,
+    bool? buildBundleIsModule,
+
+    // Build aar implementation parameters
+    String? buildAarProjectType,
+    String? buildAarTargetPlatform,
+
+    // Build apk implementation parameters
+    String? buildApkTargetPlatform,
+    String? buildApkBuildMode,
+    bool? buildApkSplitPerAbi,
+
+    // Build app bundle implementation parameters
+    String? buildAppBundleTargetPlatform,
+    String? buildAppBundleBuildMode,
+
+    // Create implementation parameters
+    String? createProjectType,
+    String? createAndroidLanguage,
+    String? createIosLanguage,
+
+    // Packages implementation parameters
+    int? packagesNumberPlugins,
+    bool? packagesProjectModule,
+    String? packagesAndroidEmbeddingVersion,
+
+    // Run implementation parameters
+    bool? runIsEmulator,
+    String? runTargetName,
+    String? runTargetOsVersion,
+    String? runModeName,
+    bool? runProjectModule,
+    String? runProjectHostLanguage,
+    String? runAndroidEmbeddingVersion,
+    bool? runEnableImpeller,
+    String? runIOSInterfaceType,
+    bool? runIsTest,
+  })  : eventName = DashEvent.commandUsageValues,
```

Comment:
this is not feedback for this PR, but hopefully once Dart has macros, we could re-write this type of a class as a macro, which would hopefully be less error-prone. As the language stands today, I can think of no better way to write this though.

Reply:
i'm not familiar with macro syntax, but i'm thinking we could have a list of strings (for key names) that we then iterate to expand the map literal you have, and we use the same string for the variable name in the if expression, the string literal (for the key), and then the variable name (for the value).

I've found from personal experience, repeating the same string three times in a row is one of the easiest mistakes to make and hardest to catch in code review.

Reply:
that having been said, your map literal *looks* right to me :)